Fix: stuck re-index when there're multiple whiteboards have the same UUID

tiensonqin opened the pull request:
This PR fixed the issue that re-index can be stuck if there're multiple whiteboard pages that have the same UUID.

Some changes:
1. re-index will continue even if some files fail to parse or load to db.
2. all the errors will be notified when re-index is finished
3. for each error, users can either click the file path to fix it themselves, or click the `Fix` button to replace the UUID.

Demo: https://www.loom.com/share/148330aac0864c279707293a3a9d9baf

pengx17 commented on src/main/frontend/handler/common/file.cljs:
```diff
@@ -39,41 +40,43 @@
+                                           (when (some? verbose) {:verbose verbose}))})]
+     (:tx (graph-parser/parse-file (db/get-db repo-url false) file content options)))))
+
+(defn whiteboard?
```
There is already one defined in `logseq.graph-parser.config` with the same implementation.

tiensonqin replied:
Thanks for spotting this out! It's fixed in [5bd9c23](https://github.com/logseq/logseq/pull/6890/commits/5bd9c23225a299811105bb449a003e0d81d6f9b6).

sprocketc commented on src/main/frontend/handler/events.cljs:
```diff
@@ -689,6 +690,59 @@
+         [:div.ml-1 "Resolved"]]
+        [:div
+         [:p
+          (str "It seems that another whiteboard file already have the ID \"" id
+               "\", you can fix it by change replace the ID in this file with another UUID.")]
```
```suggestion
          (str "It seems that another whiteboard file already has the ID \"" id
               "\". You can fix it by changing the ID in this file with another UUID.")]
```
Fixed some typos.

tiensonqin commented:
@pengx17 Thanks, I was able to reproduce the problem.

The reason is that `:block/name` has been saved in both `:block/parent` and `:block/left`, is it for backward compatibility?
Is it ok to remove it?

Also,  I think there's no need to store `:block/parent` if its parent is the page itself, we don't have any hierarchy support in whiteboards yet, right?

> there might be other cases where assertion on db transact will fail - not sure if we want to provide a hint for it as well

I added some instruments to collect those other cases in [2d8b5ba](https://github.com/logseq/logseq/pull/6890/commits/2d8b5ba529599bc6909e3cfc830e96b62a624b0f), hopefully it can help.

> the error message seems to be not quite user-friendly by showing the raw error message

Agreed that it's hard to read with the raw error message, I just changed it to only show the `message` of the error. It's still not user-friendly, any ideas are welcome!

tiensonqin commented:
@pengx17 Cool! I'll add some tests tomorrow.

logseq-cldwalker commented on deps/graph-parser/src/logseq/graph_parser/extract.cljc:
```diff
@@ -16,7 +16,8 @@
                :default [lambdaisland.glogi :as log])
-            [logseq.graph-parser.whiteboard :as gp-whiteboard]))
+            [logseq.graph-parser.whiteboard :as gp-whiteboard]
+            [medley.core :as medley]))
```
Adding new dependencies breaks existing usage of nbb-logseq for the graph-parser without some coordination with nbb-logseq. I put this in the [graph-parser readme](https://github.com/logseq/logseq/tree/master/deps/graph-parser#managing-dependencies) but happy to put the notice somewhere else too if it's helpful. Adding medley to nbb-logseq is probably not worth the effort for just one fn. For now could you just copy [the fn](https://github.com/weavejester/medley/blob/d1e00337cf6c0843fb6547aadf9ad78d981bfae5/src/medley/core.cljc#L22-L38) into graph-parser.util? If we end up using more medley fns, I can look into adding medley to nbb-logseq

tiensonqin replied:
Cool, I just copied `dissoc-in` into graph-parser.util.

tiensonqin commented:
@pengx17 I just added some unit tests in [this](https://github.com/logseq/logseq/pull/6890/commits/129d69474d96651c7165f0d3d79b650359a31c37) commit.

pengx17 commented:
> @pengx17 I just added some unit tests in [this](https://github.com/logseq/logseq/pull/6890/commits/129d69474d96651c7165f0d3d79b650359a31c37) commit.

Awesome!
